dev/com.ibm.ws.st.liberty.buildplugin.integration/src/com/ibm/ws/st/liberty/buildplugin/integration/manager/internal/AbstractLibertyManager.java

```diff
@@ -923,8 +923,16 @@ private void ensureServersStopped(List<IServer> serverList, IProgressMonitor mon
 
                 Trace.trace(Trace.INFO, "Stopping server: " + server.getName());
                 wsServer.getWebSphereServerBehaviour().stop(true, monitor);
-                if (IServer.STATE_STOPPED != server.getServerState()) {
-                    throw new Exception("Failed to stop server: " + server.getName());
+                wsServer.getWebSphereServerBehaviour().waitForServerStop(10000);
+                long endTime = System.currentTimeMillis() + 10000; // Give it a 10 second time-out
```

Comment:
Rather than making calls to a System method to get the time can it just have a total time (timeout in seconds * 4) and just loop 0 to timeout, break if stopped?

Reply:
I think it will more accurately reflect the 'timeout' if we get the actual time in seconds.  We do something like this in other places.